clusterman/draining/queue.py

```diff
@@ -243,14 +247,24 @@ def process_termination_queue(
                     up(mesos_operator_client, [hostname_ip])
                 except Exception as e:
                     logger.error(f'Failed to up {hostname_ip} continuing to terminate anyway: {e}')
+            elif host_to_terminate.scheduler == 'kubernetes':
+                logger.info(f'Host to kubernetes terminate: {host_to_terminate}')
+                terminate_host(host_to_terminate)
+                if host_to_terminate.scheduler == 'kubernetes':
+                    try:
+                        logger.info(f'Deleting kubernetes node')
+                        kube_delete_node(kube_operator_client, host_to_terminate)
```

Comment:
Could we double check if we need to kube_delete_node before terminate_host? I worry the race condition here would cause users' error.

Reply:
This ticket https://jira.yelpcorp.com/browse/CLUSTERMAN-468 specified that kube nodes are still active in scheduler even after `terminate_host`. My interpretation was that we needed to delete these nodes after the EC2 instance is terminated. Can @drmorr0 confirm this?

Reply:
Update: you're correct, when attempting to delete the node off the k8s scheduler, the system threw errors because some nodes already cease to exist after termination. I'll update this.